src/qt/optionsdialog.cpp

```diff
@@ -61,9 +61,15 @@ OptionsDialog::OptionsDialog(QWidget *parent, bool enableWallet) :
     ui->proxyPortTor->setEnabled(false);
     ui->proxyPortTor->setValidator(new QIntValidator(1, 65535, this));
 
-    connect(ui->connectSocks, SIGNAL(toggled(bool)), ui->proxyIp, SLOT(setEnabled(bool)));
-    connect(ui->connectSocks, SIGNAL(toggled(bool)), ui->proxyPort, SLOT(setEnabled(bool)));
-    connect(ui->connectSocks, SIGNAL(toggled(bool)), this, SLOT(updateProxyValidationState()));
+    // Disable and don't connect proxy checkbox interaction if proxy= is set by
+    // a param, to make it clear that removing the param / config is prerequisite to disabling.
+    if (!gArgs.GetArg("-proxy", "").empty()) {
+        ui->connectSocks->setEnabled(false);
+    } else {
+        connect(ui->connectSocks, SIGNAL(toggled(bool)), ui->proxyIp, SLOT(setEnabled(bool)));
+        connect(ui->connectSocks, SIGNAL(toggled(bool)), ui->proxyPort, SLOT(setEnabled(bool)));
+        connect(ui->connectSocks, SIGNAL(toggled(bool)), this, SLOT(updateProxyValidationState()));
```

Comment:
nit: Could use new connect syntax to avoid having to change it again in the future.

Reply:
@MarcoFalke since I'm only moving these lines, it's probably better to keep them the same. It probably makes rebasing easier regardless of whether #13529 or this is merged first.

Reply:
Since this pull has to be tested anyway, testing would also cover the transition to the new syntax. And wouldn't have to be repeated later on. Though, no strong opinion, just a nit.

Reply:
Rebased, so uses the new syntax now.